src/Compilers/CSharp/Test/Emit2/Emit/RuntimeProbing/ModuleCancellationTests.cs

```diff
@@ -0,0 +1,2291 @@
+                void G(long a, CancellationToken token) => throw null;
+                void G(CancellationToken token) => throw null;
+                void G(CancellationToken token, int a) => throw null;
+                void G(int a, int b, CancellationToken token) => throw null;
+                void G<T>(int a, CancellationToken token) => throw null;
+            }
+            """;
+
+        var verifier = CompileAndVerify(source);
+
+        AssertNotInstrumentedWithTokenLoad(verifier, "C.F");
+    }
+
+    [Theory]
+    [InlineData("ref", "")]
+    [InlineData("", "ref")]
+    [InlineData("ref readonly", "")]
+    [InlineData("", "ref readonly")]
+    [InlineData("ref", "ref readonly")]
+    [InlineData("ref readonly", "ref")]
+    public void ArgumentReplacement_Overload_ReturnType(string modifiers1, string modifiers2)
+    {
+        var source = $$"""
+            using System.Threading;
+            using System.Threading.Tasks;
+            
+            class C
+            {
+                void F(CancellationToken token)
+                {
+                    G(1);
+                }
+
+                {{modifiers1}} int G(int a) => throw null;
+                {{modifiers2}} int G(int a, CancellationToken token) => throw null;
+            }
+            """;
+
+        var verifier = CompileAndVerify(source);
+
+        AssertNotInstrumentedWithTokenLoad(verifier, "C.F");
+    }
+
+    [Fact]
+    public void ArgumentReplacement_Overload_InstanceToStatic()
+    {
+        var source = """
+            using System.Threading;
+            using System.Threading.Tasks;
+            
+            class C
+            {
+                void F(CancellationToken token)
+                {
+                    G(1);
+                }
+
+                void G(int a)
+                {
+                }
+            
+                static void G(int a, CancellationToken token) => throw null;
+            }
+            """;
+
+        var verifier = CompileAndVerify(source);
+
+        AssertNotInstrumentedWithTokenLoad(verifier, "C.F");
+    }
+
+    [Fact]
+    public void ArgumentReplacement_Overload_StaticToInstance()
+    {
+        var source = """
+            using System.Threading;
+            using System.Threading.Tasks;
+            
+            class C
+            {
+                void F(CancellationToken token)
+                {
+                    G(1);
+                }
+
+                static void G(int a)
+                {
+                }
+            
+                void G(int a, CancellationToken token) => throw null;
+            }
+            """;
+
+        var verifier = CompileAndVerify(source);
+
+        AssertNotInstrumentedWithTokenLoad(verifier, "C.F");
+    }
+
+    [Theory]
+    [CombinatorialData]
+    public void ArgumentReplacement_Overload_NonMatchingVisibility(
+        [CombinatorialValues("protected", "internal", "private protected", "internal protected")] string modifier1,
```

Comment:
>"protected", "internal", "private protected", "internal protected" [](http://example.com/codeflow?start=29&length=66)

Consider including "private" as well.

Reply:
Doesn't work since the method is in the base type.